File: app/src/main/kotlin/com/krsk/qiitareader/QRApplication.java
package com.krsk.qiitareader;

import android.app.Application;

import com.krsk.qiitareader.presentation.internal.di.HasComponent;
import com.krsk.qiitareader.presentation.internal.di.component.ApplicationComponent;
import com.krsk.qiitareader.presentation.internal.di.component.DaggerApplicationComponent;
import com.krsk.qiitareader.presentation.internal.di.module.ApplicationModule;

/**
 * Created by bl-lia on 2/2/16.
 */
public class QRApplication extends Application implements HasComponent<ApplicationComponent> {

    private ApplicationComponent component;
    private ApplicationComponent component() {
        if (component == null) {
            component = DaggerApplicationComponent.builder()
                                                    .applicationModule(new ApplicationModule())
                                                    .build();
        }

        return component;
    }

    @Override
    public ApplicationComponent getComponent() {
        return component();
    }
}

File: app/src/main/kotlin/com/krsk/qiitareader/QRApplication.kt
package com.krsk.qiitareader

import android.app.Application

import com.facebook.stetho.Stetho
import com.krsk.qiitareader.presentation.internal.di.HasComponent
import com.krsk.qiitareader.presentation.internal.di.component.ApplicationComponent
import com.krsk.qiitareader.presentation.internal.di.component.DaggerApplicationComponent
import com.krsk.qiitareader.presentation.internal.di.module.ApplicationModule

/**
 * Created by bl-lia on 2/2/16.
 */
class QRApplication : Application(), HasComponent<ApplicationComponent> {

    private val _component: ApplicationComponent by lazy {
        DaggerApplicationComponent.builder().applicationModule(ApplicationModule()).build()
    }

    override val component: ApplicationComponent
        get() = _component

    override fun onCreate() {
        super.onCreate()
        Stetho.initializeWithDefaults(this)
    }
}
